### PR TITLE
Additional metrics for the router, and some cleanup

### PR DESCRIPTION
**Pull request by pnarayanan**
- Added get related metrics and chunk put related metrics.
- Some cleanup of metrics related code.

Primary reviewer: Ming
Estimate: 30 mins.

Code built and formatted.

### CODE REVIEW

File: ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java [hunk 1]
```diff
@@ -169,9 +169,14 @@ private void onChunkOperationComplete(GetChunk chunk) {
         // chunk retrievals and channel writes will need to happen and for that, this operation needs the GetManager to
         // poll it periodically. If any exception is encountered while processing subsequent chunks, those will be
         // notified during the channel read.
-        getBlobResult = getOperationException() == null ? new GetBlobResult() : null;
-        operationCompleteCallback
-            .completeOperation(operationFuture, operationCallback, getBlobResult, getOperationException());
+        routerMetrics.getBlobOperationLatencyMs.update(time.milliseconds() - submissionTimeMs);
```
Comment (xiahome) [inline ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java]: Just to confirm that the operation latency is between the submission time and the time the first chunk received.

Comment (pnarayanan) [inline ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java]: latency is time to get the first byte of the response, and `getBlobOperationLatencyMs` tracks that.
There is another metric `getBlobOperationTotalTimeMs` which tracks the total time for the getBlob operation from submission to sending out the last byte of the response.

File: ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java [hunk 2]
```diff
@@ -150,8 +152,6 @@ public NonBlockingRouterMetrics(ClusterMap clusterMap) {
         metricRegistry.counter(MetricRegistry.name(NonBlockingRouter.class, "BlobExpiredErrorCount"));
     unknownReplicaResponseError =
         metricRegistry.counter(MetricRegistry.name(NonBlockingRouter.class, "UnknownReplicaResponseError"));
-    errorCountOnRouterClose =
```
Comment (xiahome) [inline ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java]: So this metric is removed?

**Comment by xiahome** [comment]
LGTM, will merge after all comments get confirmed.

File: ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java (hunk 2 above)
Comment (pnarayanan) [inline ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java]: I didn't see this metric getting updated anywhere.

**Comment by pnarayanan** [comment]
Addressed your comments.

**Comment by xiahome** [comment]
Merging.

**Pull request merged**